code/items/equipments/Hands.cs

```diff
@@ -0,0 +1,230 @@
+            GrabbedEntity.SetParent(null);
+            GrabbedEntity = null;
+        }
+
+        public void Update(TTTPlayer player) { }
+    }
+
+    class GrabbableRagdoll : IGrabbable
+    {
+        private PhysicsBody _handPhysicsBody;
+        private WeldJoint _holdJoint;
+
+
+        public bool IsHolding
+        {
+            get => _holdJoint.IsValid;
+        }
+
+        public GrabbableRagdoll()
+        {
+            _handPhysicsBody = PhysicsWorld.AddBody();
+            _handPhysicsBody.BodyType = PhysicsBodyType.Keyframed;
+        }
+
+        public void Grab(TTTPlayer player, TraceResult tr)
+        {
+            Transform attachment = player.GetAttachment("middle_of_both_hands")!.Value;
+            _handPhysicsBody.Position = attachment.Position;
+            _handPhysicsBody.Rotation = attachment.Rotation;
+
+            _holdJoint = PhysicsJoint.Weld
+                .From(_handPhysicsBody)
+                .To(tr.Body)
+                .Create();
+        }
+
+        public void Drop()
+        {
+            if (_holdJoint.IsValid)
+            {
+                _holdJoint.Remove();
+            }
+
+            _handPhysicsBody = null;
+        }
+
+        public void Update(TTTPlayer player)
+        {
+            Transform attachment = player.GetAttachment("middle_of_both_hands")!.Value;
+            _handPhysicsBody.Position = attachment.Position;
+            _handPhysicsBody.Rotation = attachment.Rotation;
+        }
+    }
+
+    [Library("equipment_hands")]
+    [Equipment(SlotType = SlotType.UtilityEquipment)]
+    [Hammer.Skip]
+    partial class Hands : TTTEquipment
+    {
+        public override string ViewModelPath => "";
+        public override bool CanDrop() => false;
+
+        private const float MAX_PICKUP_MASS = 205;
+        private const float MAX_PICKUP_DISTANCE = 75;
+        private Vector3 MAX_PICKUP_SIZE = new(50, 50, 50);
+
+        private IGrabbable GrabbedEntity;
+        private bool IsHoldingEntity => GrabbedEntity != null && GrabbedEntity.IsHolding;
+
+
+        public override void Spawn()
+        {
+            base.Spawn();
+
+            RenderColor = Color.Transparent;
+        }
+
+        public override void Simulate(Client client)
+        {
+            base.Simulate(client);
+
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (Owner is not TTTPlayer player)
+            {
+                return;
+            }
+
+            using (Prediction.Off())
+            {
+                if (Input.Pressed(InputButton.Attack1))
+                {
+                    TryGrabEntity(player);
+                }
+                else if (Input.Released(InputButton.Attack2))
+                {
+                    GrabbedEntity?.Drop();
+                }
+                else if (IsHoldingEntity)
+                {
+                    GrabbedEntity?.Update(player);
+                }
```

Comment:
I'd remove the `else `, otherwise the `GrabbedEntity` isn't updated in the frame the player pressed `InputButton.Attack1` but already holds an item. Doing this, you should return after both previous else-if checks.